automerge-cli/src/main.rs

```diff
@@ -38,9 +40,49 @@ enum Command {
         #[clap(long, short, default_value = "json")]
         format: ExportFormat,
 
-        /// File that contains automerge changes
-        changes_file: String,
+        /// Path that contains Automerge changes
+        #[clap(parse(from_os_str))]
+        changes_file: Option<PathBuf>,
     },
+
+    Import {
+        /// Format for input: json, toml
+        #[clap(long, short, default_value = "json")]
+        format: ExportFormat,
+
+        #[clap(parse(from_os_str))]
+        input_file: Option<PathBuf>,
+
+        /// Path to write Automerge changes to
+        #[clap(parse(from_os_str), long("out"), short("o"))]
+        changes_file: Option<PathBuf>,
+    },
+}
+
+fn open_file_or_stdin(maybe_path: Option<PathBuf>) -> Result<Box<dyn std::io::Read>> {
+    if atty::is(atty::Stream::Stdin) {
```

Comment:
Is this logic the wrong way around? It looks to me like it's rrequiring a path if stdin _is_ a tty, when we want to require a path if it isn't a tty right?

Reply:
Er, no wait I'm condused.

Reply:
I'm glad you made the exact same mistake as me! I originally wrote it backwards and was scratching my head for a whole 5 minutes.